src/Routing/RouteBuilder.php

```diff
@@ -596,17 +596,23 @@ public function redirect($route, $url, array $options = [])
      * to the `Controller\Admin\Api\` namespace.
      *
      * @param string $name The prefix name to use.
-     * @param callable $callback The callback to invoke that builds the prefixed routes.
+     * @param array|callable $params An array of routing defaults to add to each connected route.
+     *   If you have no parameters, this argument can be a callable.
+     * @param callable|null $callback The callback to invoke that builds the prefixed routes.
      * @return void
      */
-    public function prefix($name, callable $callback)
+    public function prefix($name, $params = [], callable $callback = null)
     {
+        if ($callback === null) {
+            $callback = $params;
+            $params = [];
+        }
         $name = Inflector::underscore($name);
         $path = '/' . $name;
         if (isset($this->_params['prefix'])) {
             $name = $this->_params['prefix'] . '/' . $name;
         }
-        $params = ['prefix' => $name];
+        $params = array_merge($params, ['prefix' => $name]);
```

Comment:
Are the keys always strings? Or can it also be a non-assoc array?
If it can only ever be strings, I agree, should be `$params += ['prefix' => $name];`.